src/main/java/problems/impl/TransactionalStack.java

```diff
@@ -0,0 +1,115 @@
+package problems.impl;
+
+import problems.exceptions.TransactionRolledbackException;
+
+import java.util.EmptyStackException;
+import java.util.Stack;
+
+public class TransactionalStack<T> {
+    private final Stack<T> stack = new Stack<>();
+    private final Stack<TransactionEvent<T>> transactionEvents = new Stack<>();
+    private Stack<TransactionEvent<T>> rolledbackTransactionEvents = new Stack<>();
```

Comment:
I don't understand the role of this stack. Why do you have to keep track of rolled back events?